addons/godot_pixelorama_importer/util/read_pxo_file.gd

```diff
@@ -36,24 +36,54 @@ static func read_pxo_file(source_file: String, image_save_path: String):
 	var spritesheet = Image.new()
 	spritesheet.create(size.x * frame_count, size.y, false, Image.FORMAT_RGBA8)
 
-	for i in range(project.frames.size()):
+	for i in range(frame_count):
 		var frame = project.frames[i]
 
 		# Prepare the frame image
 		var frame_img: Image = null
+		var layer := 0
 		for cel in frame.cels:
 			# Load the cel image
-			var cel_img = Image.new()
+			var cel_img := Image.new()
 			cel_img.create_from_data(size.x, size.y, false, Image.FORMAT_RGBA8, file.get_buffer(size.x * size.y * 4))
-
-			if frame_img == null:
-				frame_img = cel_img
-			else:
-				# Overlay each Cel on top of each other
-				frame_img.blend_rect(cel_img, Rect2(Vector2.ZERO, size), Vector2.ZERO)
-
-		# Add to the spritesheet
-		spritesheet.blit_rect(frame_img, Rect2(Vector2.ZERO, size), Vector2((size.x * i), 0))
+			
+			if project.layers[layer].visible:
+				var opacity: float = cel.opacity
+				if opacity == 1.0: # Avoid extra gd script calulations:
+					if frame_img == null:
+						frame_img = cel_img
+					else:
+						# Overlay each Cel on top of each other
+						frame_img.blend_rect(cel_img, Rect2(Vector2.ZERO, size), Vector2.ZERO)
+				else: # Blend with cel opacity:
+					if frame_img == null:
+						frame_img = cel_img
+						frame_img.lock()
+						for x in range(size.x):
+							for y in range(size.y):
+								var color := frame_img.get_pixel(x, y)
+								color.a *= opacity
+								frame_img.set_pixel(x, y, color)
```

Comment:
If the blend method fails, you can use premultiply_alpha instead https://docs.godotengine.org/en/stable/classes/class_image.html#class-image-method-premultiply-alpha